### PR TITLE
Added CaseIterable to Logger.Level enum

### PR DESCRIPTION
**Pull request by dokun1**

### CODE REVIEW

File: Sources/Logging/Logging.swift [hunk 1]
```diff
@@ -304,35 +304,35 @@ extension Logger {
     ///
     /// Raw values of log levels correspond to their severity, and are ordered by lowest numeric value (0) being
     /// the most severe. The raw values match the syslog values.
-    public enum Level {
+    public enum Level: String, CaseIterable {
```
Comment (weissi) [inline Sources/Logging/Logging.swift]: We can't take make this backed by `String`, that would make it slower unfortunately. I'm fine with `CaseIterable`.

If you want a `String` reprpresentation, you can add

```
extension Logger.Level {
    var uppercasedStringRepresentation: String {
        return String(describing: self).uppercased()
    }
}
```

or

```
extension Logger.Level {
    var uppercasedStringRepresentation: String {
        switch self {
            case .trace: return "TRACE"
            case .debug: return "DEBUG"
            case .info: return "INFO"
            case .notice: return "NOTICE"
            case .warning: return "WARNING"
            case .error: return "ERROR"
            case .critical: return "CRITICAL"
    }
}
```

to your package.

Comment (dokun1) [reply Sources/Logging/Logging.swift]: Good call on the extension - figured I could ask, but easy to remove.

**Comment by weissi** [comment]
@swift-server-bot test this please

**Comment by weissi** [comment]
@swift-server-bot add to whitelist

**Comment by weissi** [comment]
@swift-server-bot test this please

**Pull request merged**